csrc/device_lower/pass/allocation.cpp

```diff
@@ -258,9 +259,6 @@ class AllocationDomainSetup : private kir::IrVisitor {
         for (const auto i : c10::irange(tv->nDims())) {
           auto loop_id = tv->getLoopDomain().at(i);
           auto pt = loop_id->getParallelType();
-          if (!mayRequireAllocation(tv, loop_id)) {
```

Comment:
This would drop broadcast IDs as they wouldn't contribute to the final allocation of the tensor, but they would be still required for the analysis of `patchAllocationOfIndexedProducerTensor` as it uses `IRBFS`. Note that such broadcast IDs are eventually removed anyway at the end of this function.